android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -176,4 +162,9 @@ sealed class RemoteSettingsResult {
      * (e.g., network not available, server error).
      */
     data class NetworkFailure(val error: Exception) : RemoteSettingsResult()
+
+    /**
+     * Indicates that no changes were made
+     */
+    object NoChanges : RemoteSettingsResult()
```

Comment:
Is there an existing scenario where a consumer would care to know there were no changes? I'm just wondering if there would ever be an effect or action taken as a result of this. IMO, it seems likely that we could re-use `Success` here, even if the resulting state does not change.